compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeResolver.java

```diff
@@ -1450,7 +1449,8 @@ private void fillEffectiveType(BIntersectionType intersectionType,
         Iterator<BType> iterator = intersectionType.getConstituentTypes().iterator();
         BType effectiveType = iterator.next();
         BLangType bLangEffectiveType = bLangTypeItr.next();
-        if (effectiveType.tag == TypeTags.READONLY && iterator.hasNext()) {
+        BType effectiveTypeReferenceType = Types.getImpliedType(effectiveType);
```

Comment:
Renaming these to include ImpliedType may help with readability.

Reply:
updated in [commit](https://github.com/ballerina-platform/ballerina-lang/pull/42025/commits/5561bfc4b63f1cc862efd176700b6eba79cbbbbd).